core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1021,6 +1036,13 @@ class Log(@volatile private var _dir: File,
           // re-initialize leader epoch cache so that LeaderEpochCheckpointFile.checkpoint can correctly reference
           // the checkpoint file in renamed log directory
           initializeLeaderEpochCache()
+          if (!partitionMetadataFile.isEmpty && !partitionMetadataFile.get.isEmpty()) {
+            val partitionMetadata = partitionMetadataFile.get.read()
+            initializePartitionMetadata()
+            partitionMetadataFile.get.write(partitionMetadata.topicId)
```

Comment:
Why are we reading and writing again?

Reply:
Perhaps my understading is wrong, but if we rename the directory, do we need to create the file in the new directory? With the same topic ID and such as before.

Reply:
Oh I did understand wrong. The file is already in place, so initializePartitionMetadata is all that is needed to get the correct file path.